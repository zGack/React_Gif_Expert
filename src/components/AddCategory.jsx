import PropTypes from 'prop-types'
import {useState} from "react";

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }
    const onSubmit = (event) => {
        const newInputValue = inputValue.trim();

        event.preventDefault(); // evite que se recarge la pagina al enviar el form

        if (newInputValue.length > 1) {
            onNewCategory(newInputValue);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={onSubmit} aria-label='form'>
            <input type='text'
                placeholder="Buscar GIFs"
                value={inputValue}
                onChange={onInputChange}>
            </input>
        </form>
    );
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
