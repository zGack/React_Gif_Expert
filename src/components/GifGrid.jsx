import PropTypes from 'prop-types';
import {GifItem} from "./GifItem";
import {useFetchGifs} from "../hooks/useFetchGifs";

export const GifGrid = ({category}) => {

    const {images, isLoading} = useFetchGifs(category);

    return (
        <>
            <div>
                <h3 aria-label='h3'>{category}</h3>
            </div>

            {
                isLoading && (<h2>Cargando...</h2>)
            }

            <div className="card-grid">
                {images.map((img) => (
                    <GifItem key={img.id}
                        {...img}
                    />
                ))}
            </div>
        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
