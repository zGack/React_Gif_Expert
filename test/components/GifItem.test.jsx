import {render, screen} from "@testing-library/react";
import {GifItem} from '../../src/components/GifItem';

describe('Pruebas en <FirstApp />', () => {
    const title = 'pokemon';
    const url = 'https://pokemon.com/pikachu.jpg';

    test('debe hacer match con el snapshot', () => {
        const {container} = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GifItem title={title} url={url} />)
        //screen.debug();
        //expect(screen.getByRole('img').src).toBe(url);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt)
    })

    test('debe mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy();
    })

});
