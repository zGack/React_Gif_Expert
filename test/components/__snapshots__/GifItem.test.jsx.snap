// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="pokemon"
      src="https://pokemon.com/pikachu.jpg"
    />
    <p>
      pokemon
    </p>
  </div>
</div>
`;
